### ISSUE TITLE
GroupDocs.Viewer.dll increasing rapidly in size during the last months

### ISSUE THREAD

**Issue by ManuelHaas** [groupdocs-viewer/GroupDocs.Viewer-for-.NET-UI#42]
Hi,

GroupDocs.Viewer.dll is by far the largest DLL we have included in our application. It is now responsible for half of the total size of our application. This is now becoming a problem. 
Are there any plans to restructure the library? E.g. to split it into single functional parts. Then only those parts will be included whose functionality is actually needed. I know this from other large libraries.

GroupDocs.Viewer.dll over the last year:
22.5    (05/04/2022): 74.6 MB
22.11 (11/20/2022): 78.3 MB
23.3   (31/03/2023): 89.2 MB
23.4   (28/04/2023): 116 MB

**Comment by vladimir-litvinchik** [comment]
@ManuelHaas 

Yes, you're right, we're using fat libraries that make GroupDocs.Viewer.dll larger. We do understand that this is already an issue. We're looking for ways to reduce the assembly size and one of the ways is by splitting it into smaller pieces. Another solution that we're considering is reducing base libraries. At the moment we don't have any ETA. I'll keep this issue open to update when  we have any new information.

**Comment by vladimir-litvinchik** [comment]
@ManuelHaas 

In the latest version of [GorupDocs.Viewer for .NET 23.11](https://www.nuget.org/packages/groupdocs.viewer/23.11.0) we have managed to reduce `GroupDocs.Viewer.dll` size without any limitations on functionality.

.NET Framework DLL - size reduced by 17Mb from 115 Mb in `23.10` to 98 Mb in `23.11`.
.NET Standard assembly by - size reduced by 19Mb from 123 Mb in `23.10` to 104 Mb in `23.11`.

We are still looking for possible ways to reduce DLL size and will let you know in case of any updates.

**Comment by ManuelHaas** [comment]
@vladimir-litvinchik Thanks, that sounds great!
